nmdc_runtime/site/changesheets/base.py

```diff
@@ -0,0 +1,81 @@
+# nmdc_runtime/site/changesheets/base.py
+"""
+base.py: Provides data classes for creating changesheets for NMDC database objects.
+"""
+
+import logging
+import time
+from dataclasses import dataclass
+from pathlib import Path
+import requests
+from typing import Any, ClassVar, Dict, TypeAlias
+
+from nmdc_runtime.site.resources import RuntimeApiUserClient
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s %(levelname)s %(" "message)s"
+)
+
+JSON_OBJECT: TypeAlias = Dict[str, Any]
+CHANGESHEETS_DIR = Path(__file__).parent.absolute().joinpath("changesheets_output")
+
+
+@dataclass
+class ChangesheetLineItem:
+    """
+    A line item in a changesheet
+    """
+
+    id: str
+    action: str
+    attribute: str
+    value: str
+
+    @property
+    def line(self) -> str:
+        return f"{self.id}\t{self.action}\t{self.attribute}\t{self.value}"
```

Comment:
I am wondering whether this accounts for the possibility that the "cell" values, themselves, contains tab characters. 

I see this on [the Wikipedia page about the TSV format](https://en.wikipedia.org/wiki/Tab-separated_values#Character_escaping):

> The [IANA](https://en.wikipedia.org/wiki/Internet_Assigned_Numbers_Authority) [media type](https://en.wikipedia.org/wiki/Media_type) standard for TSV achieves simplicity by simply disallowing tabs within fields.

I don't know whether the person who wrote that meant that there is no way to represent a tab-containing value within a TSV file. 🤷 

Caveat: I haven't looked at the code that calls this function / uses this class yet.

Reply:
Based on my own _guess_ at the requirements for this: I think `raise`-ing an `ValueError` having an error message like "We do not allow values to contain tabs or newlines" would be an adequate way to handle that situation.

Reply:
What about stripping / de-tabbing `value`?  I can't think of a legitimate case where there SHOULD be a tab character in value